src/blobs.rs

```diff
@@ -7,42 +7,41 @@
 
 //! Iterator and utilities for `fileformat::Blob`.
 
-use blocks;
-use fileformat::Blob;
-use objects::OsmObj;
-use osmformat::PrimitiveBlock;
 use std::iter;
-use Result;
 
-rental! {
-    mod rent {
-        use osmformat::PrimitiveBlock;
-        use blocks;
-        #[rental]
-        pub struct OsmObjs {
-            block: Box<PrimitiveBlock>,
-            objs: blocks::OsmObjs<'block>,
-        }
+use crate::blocks::OsmObjs as OsmBlockObjs;
+use crate::fileformat::Blob;
+use crate::objects::OsmObj;
+use crate::osmformat::PrimitiveBlock;
+use crate::Result;
+
+self_cell!(
+    struct OsmBlobObjs {
+        #[from_fn]
+        owner: PrimitiveBlock,
+
+        #[covariant]
+        dependent: OsmBlockObjs,
     }
-}
+);
 
-impl<'a> Iterator for rent::OsmObjs {
+impl<'a> Iterator for OsmBlobObjs {
     type Item = OsmObj;
 
     fn next(&mut self) -> Option<Self::Item> {
-        self.rent_mut(|objs| objs.next())
+        self.with_dependent_mut(|_, objs| objs.next())
     }
 }
 
 /// An iterator on `Result<OsmObj>`.
-pub struct OsmObjs(OsmObjsImpl);
+pub struct OsmObjsIter(OsmObjsImpl);
```

Comment:
This rename is a breaking change, better to keep the old name.

Reply:
Good point, will do.